### PR TITLE
react native local evaluation

### PR DESCRIPTION
**Pull request by bgiori**
# Amplitude Developer Docs PR

## Description

SDK doc update for react native client-side local evaluation

## Deadline

(Tuesday)

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/experiment/sdks/react-native-sdk.md [hunk 1]
```diff
@@ -49,34 +49,76 @@ pod install
 
 !!!tip "Quick Start"
 
-    1. [Initialize the experiment client](#initialize)
-    2. [Fetch variants for the user](#fetch)
-    3. [Access a flag's variant](#variant)
+    The right way to initialize the Experiment SDK depends on whether you use an
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:52]:
```suggestion
    The correct way to initialize the Experiment SDK depends on whether you use an
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 2]
```diff
@@ -49,34 +49,76 @@ pod install
...
+    Amplitude SDK for analytics or a third party (e.g. Segment).
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:53]:
```suggestion
    Amplitude SDK for analytics or a third party tool like Segment.
```
Not sure why this is wrapped onto a new line.

File: docs/experiment/sdks/react-native-sdk.md [hunk 3]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
+    ```
 
-    The Amplitude Experiment SDK is set up to integrate seamlessly with the Amplitude Analytics SDK. All you need to do is update your SDK versions to the latest, and use the special integration initialization function.
+#### Configuration
 
-    ```js hl_lines="2"
-    init('<API_KEY>');
-    const experiment = Experiment.initializeWithAmplitudeAnalytics('<DEPLOYMENT_KEY>');
-    ```
+The SDK client can be configured once on initialization.
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:184]:
```suggestion
SDK client configuration occurs during initialization.
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 4]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
+    | `initialVariants` | An initial set of variants to access. This field is valuable for bootstrapping the client SDK with values rendered by the server using server-side rendering (SSR). | `{}` |
+    | `serverZone` | Select the Amplitude data center to get flags and variants from, `us` or `eu`. | `us` |
+    | `serverUrl` | The host to fetch remote evaluation variants from. For hitting the EU data center, use `serverZone`. | `https://api.lab.amplitude.com` |
+    | `flagsServerUrl` | The host to fetch local evaluation flags from. For hitting the EU data center, use `serverZone`. | `https://flag.lab.amplitude.com` |
+    | `fetchTimeoutMillis` | The timeout for fetching variants in milliseconds. | `10000` |
+    | `retryFetchOnFailure` | Whether to retry variant fetches in the background if the request doesn't succeed. | `true` |
+    | `automaticExposureTracking` | If true, calling [`variant()`](#variant) will track an exposure event through the configured `exposureTrackingProvider`. If no exposure tracking provider is set, this configuration option does nothing.  | `true` |
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:197]:
```suggestion
    | `automaticExposureTracking` | If true, calling [`variant()`](#variant) tracks an exposure event through the configured `exposureTrackingProvider`. If no exposure tracking provider is set, this configuration option does nothing.  | `true` |
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 5]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
+    | `fetchOnStart` | If true, always [fetch](#fetch) remote evaluation variants on [start](#start). If false, never fetch on start. If undefined, dynamically determine whether to fetch on start. | `undefined` |
+    | `pollOnStart` | Poll for local evaluation flag configuration updates once per minute on [start](#start). | `true` |
+    | `automaticFetchOnAmplitudeIdentityChange` | Only matters if you use the `initializeWithAmplitudeAnalytics` initialization function to seamlessly integrate with the Amplitude Analytics SDK. If `true` any change to the user ID, device ID or user properties from analytics will trigger the experiment SDK to fetch variants and update it's cache. | `false` |
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:200]:
```suggestion
    | `automaticFetchOnAmplitudeIdentityChange` | Only matters if you use the `initializeWithAmplitudeAnalytics` initialization function to seamlessly integrate with the Amplitude Analytics SDK. If `true` any change to the user ID, device ID or user properties from analytics triggers the experiment SDK to fetch variants and update it's cache. | `false` |
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 6]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
 
-    Using the integration initializer will automatically configure implementations of the [user provider](#user-provider) and [exposure tracking provider](#exposure-tracking-provider) interfaces to pull user data from the Amplitude Analytics SDK and track exposure events.
+!!!info "EU Data Center"
+    If you're using Amplitude's EU data center, configure the `serverZone` option on initialization to `eu`.
 
-???segment "Segment Integration"
+#### Integrations
 
-    Experiment's integration with Segment Analytics is still a manual implementation at this point. Copy the exposure tracking provider implementation into your app code base and initialize the Experiment SDK with the provider instances in the configuration.
+If you use either Amplitude or Segment Analytics SDKs to track events into Amplitude, you'll want to set up an integration on initialization. Integrations automatically implement [provider](#providers) interfaces to enable a more streamlined developer experience by making it easier to **manage user identity** and **track exposures events**.
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:210]:
```suggestion
If you use either Amplitude or Segment Analytics SDKs to track events into Amplitude, Amplitude recommends that you set up an integration on initialization. Integrations implement [provider](#providers) interfaces to enable a more streamlined developer experience by making it easier to **manage user identity** and **track exposures events**.
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 7]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
+        user_id: analytics.user().id(),
+        device_id: analytics.user().analyticsId(),
     });
     ```
 
-#### Configuration
+### Start
 
-The SDK client can be configured once on initialization.
+Start the SDK by getting flag configurations from the server and fetching remote evaluation variants for the user. The SDK is ready once the returned promise resolves.
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:253]:
```suggestion
The Experiment SDK starts when you get flag configurations from the server and fetch remote evaluation variants for the user. The SDK is ready after the returned promise resolves.
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 8]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
-    | `exposureTrackingProvider` | Implement and configure this interface to track exposure events through the experiment SDK, either automatically or explicitly. | `null` |
-    | `instanceName` | Custom instance name for experiment SDK instance. **The value of this field is case-sensitive.** | `null` |
+```js
+start(user?: ExperimentUser): Promise<void>
+```
 
----
+| Parameter | Requirement | Description |
+| --- | --- | --- |
+| `user` | optional | Explicit [user](../general/data-model.md#users) information to pass with the request to fetch variants. This user information is merged with user information provided from [integrations](#integrations) via the [user provider](#user-provider), preferring properties passed explicitly to `fetch()` over provided properties. Also sets the user in the SDK for reuse. | `undefined` |
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:261]:
```suggestion
| `user` | optional | Explicit [user](../general/data-model.md#users) information to pass with the request to fetch variants. This user information merges with user information from any [integrations](#integrations) through the [user provider](#user-provider), and prefers properties passed explicitly to `fetch()` over provided properties. Also sets the user in the SDK for reuse. | `undefined` |
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 9]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
 
-!!!info "EU Data Center"
-    If you're using Amplitude's EU data center, configure the `serverUrl` option on initialization to `https://api.lab.eu.amplitude.com`
+Call `start()` when your application is initializing, after user information is available to use to evaluate or [fetch](#fetch) variants. The returned promise resolves after loading local evaluation flag configurations and fetching remote evaluation variants.
+
+Configure the behavior of `start()` by setting `fetchOnStart` in the SDK configuration on initialization to **improve performance based on the needs of your application**.
+
+* If your application always relies on remote evaluation on startup, set `fetchOnStart` to `true` to load flag configurations and fetch remote evaluation variants simultaneously.
+* If your application never relies on remote evaluation, set `fetchOnStart` to `false` to avoid increased startup latency if someone accidentally creates a remote evaluation flag.
+* If your application relies on remote evaluation, but not right at startup, you may set `fetchOnStart` to `false` and call `fetch()` and await the promise separately.
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:269]:
```suggestion
* If your application relies on remote evaluation, but not right at startup, set `fetchOnStart` to `false` and call `fetch()` and await the promise separately.
```

File: docs/experiment/sdks/react-native-sdk.md [hunk 10]
```diff
@@ -90,102 +132,160 @@ The following functions make up the core of the Experiment client-side SDK.
...
----
+| Parameter | Requirement | Description |
+| --- | --- | --- |
+| `user` | optional | Explicit [user](../general/data-model.md#users) information to pass with the request to fetch variants. This user information is merged with user information provided from [integrations](#integrations) via the [user provider](#user-provider), preferring properties passed explicitly to `fetch()` over provided properties. Also sets the user in the SDK for reuse. | `undefined` |
 
-!!!info "EU Data Center"
-    If you're using Amplitude's EU data center, configure the `serverUrl` option on initialization to `https://api.lab.eu.amplitude.com`
+Call `start()` when your application is initializing, after user information is available to use to evaluate or [fetch](#fetch) variants. The returned promise resolves after loading local evaluation flag configurations and fetching remote evaluation variants.
+
+Configure the behavior of `start()` by setting `fetchOnStart` in the SDK configuration on initialization to **improve performance based on the needs of your application**.
```
Comment (markzegarelli) [inline docs/experiment/sdks/react-native-sdk.md:265]:
```suggestion
Set `fetchOnStart` in the SDK configuration to set the behavior of `start()` to improve the performance of your application.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: docs/experiment/sdks/react-native-sdk.md [hunk 11]
```diff
@@ -49,34 +49,76 @@ pod install
...
+            // Flag is on
+        } else {
+            // Flag is off
+        }
+        ```
+
+    === "Third Party Analytics"
+
+        4. [Initialize the experiment client](#initialize)
+        5. [Start the SDK with the user](#fetch)
```
Comment (bgiori) [inline docs/experiment/sdks/react-native-sdk.md:84]:
```suggestion
        5. [Start the SDK with the user](#start)
```

File: docs/experiment/sdks/react-native-sdk.md (hunk 7 above)
Comment (bgiori) [reply docs/experiment/sdks/react-native-sdk.md:253]:
```suggestion
Start the Experiment SDK to get flag configurations from the server and fetch remote evaluation variants for the user. The SDK is ready once the returned promise resolves.
```

**Review by markzegarelli — APPROVED** [review]

**Pull request merged**